### DISCUSSION TITLE
.whl support error

### DISCUSSION THREAD

**Discussion by WidderGit** [thliebig/openEMS-Project#221] [General]
Hello,
I'm attempting to install the CSXCAD and OpenEMS .whl files but I am getting an error ".whl is not a supported wheel on this platform".  Can anyone offer a suggestion to resolve this error?
Thanks for your time
Marc

**Comment by WidderGit** [comment]
I have some information to add.
I am running Python 3.12.1.  It looks like the latest .wrl available for CSXCAD and openEMS is 3.11, so perhaps all I need is an updated set of .whl files.  Unfortunately, I am not sure how to do this...
Marc

**Comment by LubomirJagos42** [comment]
hi, that CSXCAD wheel is written in cython what python with datatypes (simply said) and therefore it's interfaceing some DLL when it's written in python 3.11 you need to run it in 3.11 in your case you just need install pyhon 3.11 and install it into python 3.11

Steps:
  1. install python 3.11
  2. use command [python dir]/Scripts/pip3 install <wheel file name>
  
Then you need to just run python 3.11 when using openEMS python interface.

**Reply by WidderGit** [reply]
I was worried that was the case, but I went ahead and took your advice to get past this particular step.  Still working through the next steps.
Thanks
Marc

**Reply by LubomirJagos42** [reply]
you will sooner or later hit some other problems with python 3.12, ie. I was working on some addon for openEMS for FreeCAD and it's also running in Blender, it's GUI for openEMS and it's using Pyside2 library for Qt5 and when I installed newest Blender there is python 3.12 and was looking how to run it and figure out there is no Pyside2 support in 3.12, there is I think Pyside6 for Qt6 what is no problem but it would need modification and FreeCAD is using 3.8 version.
All in all I would recommend to use python 3.11 now over 3.12 if you don't really need it since lot of stuff which is already written possibly will not run if there are newer versions of libraries or you must in case of cython recompile things, cython things are more sensible since if there is cython DLL for 3.11 python you can't use 3.12, not sure if I wrote it right but this limitation is due cython.
I don't see this as problem since you can have on same PC as many pythons installed as you want :D :D you just have to run the right one :D

**Reply by LubomirJagos42** [reply]
welcome to dependency hell :)

**Comment by WidderGit** [comment]
Yes, it looks like I will be able to bypass all the 3.12 troubles as I downloaded 3.11.
I'm now stuck at the next step in the instructions.  
The first setup at \...\CSXCAD\Python appears to have worked but when I CD down to C:\openEMS_v0.0.36\python
I get this:
python: can't open file 'C:\\openEMS_v0.0.36\\openEMS\\python\\setup.py': [Errno 2] No such file or directory

Note that the root of my openEMS install is openEMS_v0.0.36 instead of opt (I don't think that matters).  

I think I am probably missing something obvious, but I am not experienced with this sort of thing. Any advice is appreciated.

Thanks
Marc

**Reply by WidderGit** [reply]
Disregard the question.  there at least 2 paths containing openEMS/Python, I was in the path closer to the root rather than the one under openEMS-Project-master
Marc

**Reply by WidderGit** [reply]
Update
I resolved the issue above, it was due to there being 2 paths that are identical.
Working on the next error
Marc
